src/util.cpp

```diff
@@ -108,6 +108,33 @@ std::string get_process_comm(pid_t pid)
     }
 }
 
+std::vector<std::string> get_process_cmdline(pid_t pid)
+{
+    try
+    {
+        std::vector<std::string> cmdline;
+        boost::filesystem::ifstream s{ boost::filesystem::path{ "/proc" } / std::to_string(pid) /
+                                       "cmdline" };
+
+        // /proc/<pid>/cmdline is a NUL-delimited list of arguments
+        for (std::string arg; std::getline(s, arg, '\0');)
+        {
+            Log::debug() << "get_process_cmdline: arg=" << arg;
+            cmdline.emplace_back(std::move(arg));
```

Comment:
I don't trust that move... like at all. It scares me. 😨

Reply:
I concede that this may look frightening, but it should actually be perfectly fine.

`std::getline` calls `arg.erase()` before accessing `arg` in any other way. When moved-from, `std::string`s are in a 'valid, but unspecified state', but `std::string::erase()` has no preconditions, therefore it does not invoke undefined when called on a moved-from object.<sup>[[1]](https://en.cppreference.com/w/cpp/utility/move)</sup>

Reply:
Okay, but why not use something in that direction:
```
while(std::getline(s, cmdline.back(),'\0')) {
  // ...
  cmdline.resize(cmdline.size() + 1);
}
```

this would also get rid of the unconventional for-loop-header

Reply:
But this will fail on the first iteration, won't it :thinking: Since `cmdline` is empty then and calling `.back()` on it surely is ub?

I'm really not fond of the idea of explicitly modifying `cmdline` (e.g. with `.resize()`) as long as there's `.emplace_back()`. And I don't think we can (or should) eliminate the loop variable. Imo compilers are smart, this function is probably not performance critical and the variable `arg` kind of adds to the expressiveness of this loop.

If your gripe is with the unusual `for`-loop header, one can always do:
```cpp
for (std::string arg;
     std::getline(s, arg, '\0');
     cmdline.emplace_back(std::move(arg)))
    ;
```
which follows the 'initialize-check-update' pattern of the `for`-loop or
```cpp
std::string arg;
while (std::getline(s, arg, '\0')) {
    cmdline.emplace_back(std::move(arg)));
}
```
(though this adds `arg` in the surrounding scope).

Reply:
Why I don't like this is because of readability. I'm thinking about some clever combination of back_insterter and std::copy, but I don't think it will directly work this way:

```
std::copy(std::istream_iterator<std::string>(s), std::istream_iterator<std::string>(), std::back_inserter(cmdline));
```